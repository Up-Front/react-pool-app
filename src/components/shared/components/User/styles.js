import styled from 'styled-components';

export const UserWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  background: 'white';
  padding: 10px;
  border-radius: 8px;
`;

export const UserEloRating = styled.div``;
