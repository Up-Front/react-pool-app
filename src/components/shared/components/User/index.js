import React from 'react';
import PropTypes from 'prop-types';
import Avatar from './../Avatar';
import RankingDetail from './../RankingDetail';
import { UserWrapper, UserEloRating } from './styles';

const User = ({ user, ...props }) => {
  const handleClickEvent = () => {
    if (typeof props.handleClick === 'function') {
      props.handleClick(props.uid, user);
    }
  };

  return (
    <UserWrapper onClick={handleClickEvent}>
      <RankingDetail
        currentRanking={user.currentRanking}
        previousRanking={user.previousRanking}
      />
      <Avatar user={user} />
      <div>
        <strong>{user.displayName}</strong>
        <UserEloRating data-test="elorating">
          points: <span>{user.currentRanking.eloRating}</span>
        </UserEloRating>
      </div>
    </UserWrapper>
  );
};
User.propTypes = {
  handleClick: PropTypes.func,
  user: PropTypes.object.isRequired,
};

export default User;
