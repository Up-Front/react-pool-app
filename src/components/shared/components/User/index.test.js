import React from 'react';
import { shallow } from 'enzyme';
import User from './';

describe('User component', () => {
  let uid;
  let auth;
  let handleClick;
  beforeEach(() => {
    uid = '1';
    auth = {
      uid: 1,
      displayName: 'Bob de Bouwer',
      avatarUrl: 'http://image',
      currentRanking: {},
    };

    handleClick = jest.fn();
  });

  test('renders without crashing', () => {
    const wrapper = shallow(<User uid={uid} user={auth} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders check click handle', () => {
    const wrapper = shallow(
      <User uid={uid} user={auth} handleClick={handleClick} />
    );

    wrapper.find('[onClick]').simulate('click');
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
