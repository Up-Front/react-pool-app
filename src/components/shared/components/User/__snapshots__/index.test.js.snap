// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <User
    uid="1"
    user={
        Object {
            "avatarUrl": "http://image",
            "displayName": "Bob de Bouwer",
            "uid": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <RankingDetail
          currentRanking={Object {}}
          previousRanking={Object {}}
/>,
        <Avatar
          user={
                    Object {
                              "avatarUrl": "http://image",
                              "displayName": "Bob de Bouwer",
                              "uid": 1,
                            }
          }
/>,
        <div>
          <strong>
                    Bob de Bouwer
          </strong>
          <div />
</div>,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentRanking": Object {},
          "previousRanking": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "user": Object {
            "avatarUrl": "http://image",
            "displayName": "Bob de Bouwer",
            "uid": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              Bob de Bouwer
</strong>,
            <div />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Bob de Bouwer",
            },
            "ref": null,
            "rendered": "Bob de Bouwer",
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <RankingDetail
            currentRanking={Object {}}
            previousRanking={Object {}}
/>,
          <Avatar
            user={
                        Object {
                                    "avatarUrl": "http://image",
                                    "displayName": "Bob de Bouwer",
                                    "uid": 1,
                                  }
            }
/>,
          <div>
            <strong>
                        Bob de Bouwer
            </strong>
            <div />
</div>,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentRanking": Object {},
            "previousRanking": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "user": Object {
              "avatarUrl": "http://image",
              "displayName": "Bob de Bouwer",
              "uid": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Bob de Bouwer
</strong>,
              <div />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Bob de Bouwer",
              },
              "ref": null,
              "rendered": "Bob de Bouwer",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
